ecleankernel/layout/blspec.py

```diff
@@ -128,11 +128,18 @@ def find_kernels(self,
         # collect from ESP/Linux
         if self.ukidir.is_dir():
             for file in os.listdir(self.ukidir):
-                if (not file.startswith(self.kernel_id) or
-                        not file.endswith(".efi")):
-                    # This file is not an efi file or does not belong to us
+                if not file.endswith(".efi"):
+                    # Not an UKI
                     continue
-                ver = file.removeprefix(self.kernel_id +
+                if file.startswith(self.kernel_id):
```

Comment:
How about something like:

```
ver = file.removeprefix(f"{self.kernel_id}-").removeprefix("gentoo-")
if file == ver:
    continue
```

(technically it would also eat duplicate kernel_id+gentoo but I think we can ignore that)

Reply:
Sure, that also works. We do still have to account for the `.efi` suffix as well though.